iree/test/e2e/regression/generate_e2e_matmul_tests.py

```diff
@@ -533,7 +674,15 @@ def main(args):
   lhs_rhs_type = MatrixElemTypeId(args.lhs_rhs_type)
   acc_type = infer_acc_type(lhs_rhs_type)
   shapes_id = ShapesId(args.shapes)
-  (function_definitions, traces) = generate(lhs_rhs_type, acc_type, shapes_id)
+  compilation_info_id = CompilationInfoId(args.compilation_info)
+  if compilation_info_id == CompilationInfoId.LLVMGPUMatmulSimt or compilation_info_id == CompilationInfoId.LLVMGPUMatmulTensorCore:
+    (function_definitions, traces) = generate_gpu(lhs_rhs_type, acc_type,
+                                                  shapes_id,
+                                                  compilation_info_id)
+  else:
```

Comment:
An error should be generated if `compilation_info_id` is not one of the expected values. Otherwise, my undertanding is that at the moment, if one makes a typo in the `compilation_info` flag, that silently falls through to the below `generate_default` path.  If you accept my above suggestion to have the default value be `""` instead of `"none"` then that would be a matter of checking `if compilation_info_id:` .